### DISCUSSION TITLE
109 → 110 regression in `wasm-emscripten-finalize`

### DISCUSSION THREAD

**Discussion by shaded-enmity** [WebAssembly/binaryen#5056] [Q&A] [answered]
It seems that in version 110 `wasm-emscripten-finalize` stopped printing information to `stdout` which is subsequently consumed by Emscripten (2.0.17 - I'm hard locked to this version) which causes an error.

In 109 the output looked something like this:

```
  metadata = {
    'declares': [],
    'externs': [],
    'staticBump': 0,
    'tableSize': 0,
    'exports': [],
    'namedGlobals': {},
    'emJsFuncs': {},
    'asmConsts': {},
    'invokeFuncs': [],
    'features': [],
    'mainReadsParams': 1,
    'globalImports': []
  }
```

But in 110 the output is empty. Are there any command line options I can use to make it output the above metadata in JSON format or do I need to go back to 109?

**Comment by tlively** [comment, marked as answer]
Emscripten and Binaryen are version-locked, so we don't try to maintain any compatibility between Emscripten and Binaryen versions other than the one it ships with. For best results you should use the exact version of Binaryen that ships with your version of Emscripten via emsdk, which in general will not be a tagged release version of Binaryen.

**Reply by shaded-enmity** [reply]
Thank you, good to know, so back to 109 it is :)
